airbyte-cdk/bulk/toolkits/load-s3/src/main/kotlin/io/airbyte/cdk/load/file/s3/S3MultipartUpload.kt

```diff
@@ -157,36 +157,48 @@ class S3StreamingUpload(
     private val response: CreateMultipartUploadResponse,
 ) : StreamingUpload<S3Object> {
     private val log = KotlinLogging.logger {}
-    private val uploadedParts = ConcurrentLinkedQueue<CompletedPart>()
+    private val uploadedParts = ConcurrentHashSet<CompletedPart>()
+    private val isComplete = AtomicBoolean(false)
+
+    override suspend fun uploadPart(part: ByteArray, index: Int) {
```

Comment:
Nit: Can we call this a part number instead of index? Or is it actually an index to something?